source/common/runtime/runtime_impl.cc

```diff
@@ -425,7 +425,11 @@ void DiskLayer::walkDirectory(const std::string& path, const std::string& prefix
   }
 
   Filesystem::Directory directory(path);
-  for (const Filesystem::DirectoryEntry& entry : directory) {
+  Filesystem::DirectoryIteratorImpl it = directory.begin();
+  RELEASE_ASSERT(it.status().ok(), it.status().ToString());
```

Comment:
Sorry this was a fatal failure but it should probably be a non-fatal exception going forward.  THROW_IF_NOT_OK instead?
/wait